Tests/AK/TestMemory.cpp

```diff
@@ -66,3 +68,12 @@ TEST_CASE(kmp_two_chunks)
     EXPECT_EQ(result_2.value_or(9), 4u);
     EXPECT(!result_3.has_value());
 }
+
+TEST_CASE(timingsafe_compare)
```

Comment:
Idea for a future "make sure we don't branch on input data" I've had for a while that seems relevant here:
a memory "poison" API for UE (set all shadow bits to 0), plus an option to make UE quit on such branches; then `poison(a); posion(b); timing_safe_compare(a, b);` should cleanly execute (your current impl - as noted by the fixme - _can_ fail this test)

Reply:
I like that idea, not sure I'll ever get around to it, but certainly sounds useful.

Reply:
lol same here, it'll happen eventually when we need it I guess